patroni/postgresql/__init__.py

```diff
@@ -407,6 +407,12 @@ def start(self, timeout=None, task=None, block_callbacks=False, role=None):
         self._pending_restart = False
 
         configuration = self.config.effective_configuration
+        if "unix_socket_directories" in configuration:
+            for d in configuration["unix_socket_directories"].split(","):
+                if not d.startswith('/'):
+                    d = os.path.join(self._data_dir, d)
+                if not os.path.isdir(d):
+                    os.makedirs(d)
```

Comment:
`os.makedirs()` will fail if there is already something...

Reply:
Well, if there is already something but directory, the server won't start anyway, and I don't think we should resolve that automatically, the only thing we can do about it, make an additional check and give the user a hint.

Reply:
added a try/except